dts/arm/st/g0/stm32g0.dtsi

```diff
@@ -20,6 +20,10 @@
 		zephyr,flash-controller = &flash;
 	};
 
+	aliases {
+		gpio-controller-parent = &pinctrl;
+	};
+
```

Comment:
This looks like a bad workaround:
- `&pinctrl` is de facto parents of gpio-controllers. 
- aliases are generally reserved for application purpose. Maybe `chosen` would be a better match, but I'm not sure.
- since gpio-controllers actually define a `gpio-controller` property, we should be able to get rid of this using DT_PARENT

This being said, it actually seems we miss a piece in DT machinery to identify a node based on its properties.
@mbolivar-nordic How do you feel about this proposal ?

Reply:
I agree that the workaround is problematic, and happy to hear about better suggestions.
chosen may be the better option, and can make that change if that is the best approach.
I did struggle finding a good consistent solution here that worked across all cases.
I am not sure how DT_PARENT can be used here? Can you expand on that?

Reply:
> I am not sure how DT_PARENT can be used here? Can you expand on that?

Once you get a gpio node, you can use DT_PARENT() to access `&pinctrl` node

Reply:
Ah,  that's the trick :-) getting the GPIO node (or the parent) is the hard bit to do in a platform independent way.